echidna/limit/limit_setting.py

```diff
@@ -0,0 +1,202 @@
+            self._roi = kwargs.get("roi")
+        else:
+            self._roi = None
+
+        # Check spectra are compatible
+        for background in backgrounds:
+            assert (background._energy_width == signal._energy_width), \
+                "cannot compare histograms, different energy bin width"
+            assert (background._radial_width == signal._radial_width), \
+                "cannot compare histograms, different radial bin width"
+            assert (background._time_width == signal._time_width), \
+                "cannot compare histograms, different time bin width"
+
+    def configure_signal(self, signal_config):
+        """ Supply a configuration object associated with the signal
+
+        Args:
+          signal_config (:class: `limit_config.LimitConfig`): signal
+            configuration
+        """
+        self._signal_config = signal_config
+
+    def configure_background(self, name, background_config):
+        """ Supply a configuration object associated with the background
+
+        Args:
+          background_config (:class: `limit_config.LimitConfig`): 
+            background configuration
+        """
+        self._background_configs[name] = background_config
+
+    def set_calculator(self, calculator):
+        """ Sets the chi squared calculator to use for limit setting
+
+        Args:
+          calculator (:class: `chi_squared.ChiSquared`): chi squared
+            calculator to use for limit setting
+        """
+        self._calculator = calculator
+
+    def get_limit(self, limit_chi_squared=2.71):
+        """ Get signal counts at limit
+
+        Args:
+          limit_chi_squared (float): chi squared required for limit
+        """
+        assert (self._signal_config is not None), \
+            "signal limit configuration not set"
+        assert (len(self._background_configs) == len(self._backgrounds)), \
+            "missing limit configuration for one or more backgrounds"
+        assert (self._calculator is not None), \
+            "chi squared calculator not set"
+        if self._data is None:
+            observed = numpy.zeros(
+                shape=[self._signal._energy_bins],
+                dtype=float)
+            for background in self._backgrounds:
+                config = self._background_configs.get(background._name)
+                background.scale(config._prior_count)
+                observed += background.project(0)
+            self._observed = observed
+        else:  # _data is not None
+            self,_observed = self._data
+        self._signal_config.reset_chi_squareds()
+        for signal_count in self._signal_config.get_count():
+            print "LimitSetting.get_limit: signal_count =", signal_count
+            self._signal.scale(signal_count)
+            self._signal_config.add_chi_squared(
+                self._get_chi_squared(
+                    self._backgrounds, len(self._backgrounds)
+                    )
```

Comment:
Agreed! Wanted to make the function call in the argument of the first function clear, and avoid the line being too long, but this was not the way to do it! Change to:

```Python
self._signal_config.add_chi_squared(
    self._get_chi_squared(self._backgrounds,
                          len(self._backgrounds)))
```

which is more readable and hopefully looks much nicer too!